include/victoria_navigation/solve_robomagellan.h

```diff
@@ -0,0 +1,162 @@
+		MOVE_FROM_CONE,			// Move away from the cone, setting up to move to the next waypoint.
+		ADVANCE_TO_NEXT_POINT	// Advance to the next waypoint.
+	} STATE;
+
+	// Parameters.
+	string fix_topic_name_;				// Topic name containing fix message.
+	string waypoint_yaml_path_;			// Path to yaml file containing waypoints.
+
+	// Subscribers.
+	ros::Subscriber fix_sub_;
+
+	// Algorithm variables.
+	ros::ServiceClient coneDetectorAnnotatorService_;	// For annotating the cone detector image.
+	vector<GPS_POINT> gps_points_;				// Ordered list of waypoints to traverse.
+	long int index_next_point_to_seek_;			// Index into list of GPS points to seek to next.
+	victoria_perception::AnnotateDetectorImage annotator_request_;	// The annotation request.
+	STATE state_;								// State of state machine.
+	YAML::Node waypoints_;						// Waypoints in yaml format.
+	
+	// ROS node handle.
+	ros::NodeHandle nh_;
+
+	// Process one Fix topic message.
+	long int count_Fix_msgs_received_;
+	sensor_msgs::NavSatFix last_Fix_msg_;
+	void fixCb(const sensor_msgs::NavSatFixConstPtr& msg);
+
+	static double degrees(double x) { return x * 180.0 / M_PI; }
+	static double radians(double x) { return x * M_PI / 180.0; }
+
+ 	// Calculate bearing between two GPS points--accurate for distances for RoboMagellan.
+	double bearing(sensor_msgs::NavSatFix from, sensor_msgs::NavSatFix to) {
+	    double lat1 = radians(from.latitude);
+	    double lon1 = radians(from.longitude);
+	    double lat2 = radians(to.latitude);
+	    double lon2 = radians(to.longitude);
+	    double dLon = lon2 - lon1;
+	 
+	    double y = sin(dLon) * cos(lat2);
+	    double x = cos(lat1) * sin(lat2) - (sin(lat1) * cos(lat2) * cos(dLon));
+	 
+	    return degrees(atan2(y, x)); 
+	}
+	 
+	 
+ 	// Calculate distance between two GPS points--accurate for distances for RoboMagellan.
+	double distance(sensor_msgs::NavSatFix from, sensor_msgs::NavSatFix to) {
+	    double lat1 = radians(from.latitude);
+	    double lon1 = radians(from.longitude);
+	    double lat2 = radians(to.latitude);
+	    double lon2 = radians(to.longitude);
+	    double dLat = lat2 - lat1;
+	    double dLon = lon2 - lon1;
+	 
+	    double a = (sin(dLat / 2.0) * sin(dLat / 2.0)) + 
+	               ((cos(lat1) * cos(lat2)) *
+	                (sin(dLon / 2.0) * sin(dLon / 2.0)));
+	    double c = 2.0 * atan2(sqrt(a), sqrt(1 - a));
+	    
+	    return 6371000.0 * c;
+	}
+
+	// Normalize an Euler angle into [0..360).
+	double normalizeEuler(double yaw_degrees);
+
+	// Reset goal. After this, someone must request the goal again and it will start over.
+	void resetGoal();
+
+	// Singleton pattern.
+	SolveRobomMagellan();
+	SolveRobomMagellan(SolveRobomMagellan const&) {};
+	SolveRobomMagellan& operator=(SolveRobomMagellan const&) {};
+
+public:
+	RESULT_T tick();
+
+	string goalName() { return "/strategy/solve_robomagellan"; }
+
+	string name() { return string("SolveRobomMagellan"); };
+
+	static SolveRobomMagellan& singleton();
```

Comment:
Yeah, I feel like this would be much better solved without singletons using a launch file to replace `robo_magellan_node.cpp` and action messages between nodes, where `solve_robomagellan`, `seek_to_gps`, etc are all nodes. I'm don't think the speed improvement is worth the tightly coupledness and lack of introspection.
We can talk about this in the future.